### PR TITLE
Abort SigHelp if navigation keys before presented

### PR DESCRIPTION
**Pull request by rchande**

### CODE REVIEW

**Comment by rchande** [comment]
test eta

**Comment by rchande** [comment]
@dotnet/roslyn-ide please review.

File: src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Presentation/SignatureHelpPresenter.SignatureHelpPresenterSession.cs [hunk 1]
```diff
@@ -31,6 +31,14 @@ private class SignatureHelpPresenterSession : ForegroundThreadAffinitizedObject,
             private ISignatureHelpSession _editorSessionOpt;
             private bool _ignoreSelectionStatusChangedEvent;
 
+            public bool EditorSessionIsActive
```
Comment (brettfo) [inline src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Presentation/SignatureHelpPresenter.SignatureHelpPresenterSession.cs]: `public bool EditorSessionIsActive => _editorSessionOpt?.IsDismissed == false;`

File: src/EditorFeatures/Test2/IntelliSense/SignatureHelpControllerTests.vb [hunk 2]
```diff
@@ -94,6 +94,40 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
         End Sub
 
         <WpfFact>
+        Public Sub DownKeyShouldNotBlockOnModelComputation()
```
Comment (Pilchie) [inline src/EditorFeatures/Test2/IntelliSense/SignatureHelpControllerTests.vb]: Do we need equivalent tests that verify that we *do* block once it has been presented?

**Comment by Pilchie** [comment]
:+1: pending test question.

**Comment by natidea** [comment]
@dotnet-bot retest prtest/win/dbg/eta please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_prtest_win_dbg_eta/3978/
// Retest reason: Retry tests that failed due to missing binding redirects and dependencies. Those changes have been reverted.

**Comment by davkean** [comment]
retest eta please

**Comment by rchande** [comment]
@Pilchie added a test
@brettfo done.

**Comment by Pilchie** [comment]
Thanks LGTM

**Comment by natidea** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
@dotnet-bot retest prtest/win/dbg/unit64 please
// Retest reason: Failure tracked by #8807 Tests Disabled by #8822

**Comment by davkean** [comment]
@natidea Due the way that GitHub creates what it calls the "merge branch", you need to close and reopen PRs to pick up newer commits in the destination branch.

**Comment by davkean** [comment]
@natidea It's surprising at first when you learn this.

**Comment by natidea** [comment]
@davkean Hmm this one picked up the commit and the test passed

**Comment by davkean** [comment]
Hmm figuring out why/how.

File: src/EditorFeatures/Test2/IntelliSense/SignatureHelpControllerTests.vb [hunk 3]
```diff
@@ -94,6 +93,76 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
...
+                         End Function)
+
+
+            Dim handled = controller.TryHandleDownKey()
+
+            Assert.False(handled)
+        End Sub
+
+        <WpfFact>
+        Public Sub UpKeyShouldNotBlockOnModelComputation()
```
Comment (balajikris) [inline src/EditorFeatures/Test2/IntelliSense/SignatureHelpControllerTests.vb]: Nit, but is there a bug number to be associated with this?

**Comment by balajikris** [comment]
:+1:

**Comment by brettfo** [comment]
:shipit:

**Pull request merged**